cleanrl/experiments/impl_matters/ppo_continuous_gae.py

```diff
@@ -0,0 +1,493 @@
+                       help="the surrogate clipping coefficient")
+    parser.add_argument('--update-epochs', type=int, default=100,
+                        help="the K epochs to update the policy")
+    # MODFIED: Added support for KL Bounding during the updates
+    parser.add_argument('--kl', action='store_true',
+                        help='If toggled, the policy updates will be early stopped w.r.t target-kl')
+    parser.add_argument('--target-kl', type=float, default=0.015)
+
+    # MODIFIED: Separate learning rate for policy and values, according to OpenAI SpinUp
+    parser.add_argument('--policy-lr', type=float, default=3e-4)
+    parser.add_argument('--value-lr', type=float, default=1e-3)
+
+    # MODIFIED: Parameterization for the tricks in the Implementation Matters paper.
+    parser.add_argument('--norm-obs', action='store_true',
+                        help="Toggles observation normalization")
+    parser.add_argument('--norm-rewards', action='store_true',
+                        help="Toggles rewards normalization")
+    parser.add_argument('--norm-returns', action='store_true',
+                        help="Toggles returns normalization")
+    parser.add_argument('--obs-clip', type=float, default=10.0,
+                        help="Value for reward clipping, as per the paper")
+    parser.add_argument('--rew-clip', type=float, default=5.0,
+                        help="Value for observation clipping, as per the paper")
+    parser.add_argument('--anneal-lr', action='store_true',
+                        help="Toggle learning rate annealing for policy and value networks")
+    parser.add_argument('--weights-init', default="xavier", choices=["xavier", 'orthogonal'],
+                        help='Selects the scheme to be used for weights initialization')
+
+    args = parser.parse_args()
+    if not args.seed:
+        args.seed = int(time.time())
+
+# TRY NOT TO MODIFY: setup the environment
+experiment_name = f"{args.gym_id}__{args.exp_name}__{args.seed}__{int(time.time())}"
+writer = SummaryWriter(f"runs/{experiment_name}")
+writer.add_text('hyperparameters', "|param|value|\n|-|-|\n%s" % (
+        '\n'.join([f"|{key}|{value}|" for key, value in vars(args).items()])))
+
+if args.prod_mode:
+    import wandb
+    wandb.init(project=args.wandb_project_name, entity=args.wandb_entity, tensorboard=True, config=vars(args), name=experiment_name, monitor_gym=True)
+    writer = SummaryWriter(f"/tmp/{experiment_name}")
+    wandb.save(os.path.abspath(__file__))
+
+# TRY NOT TO MODIFY: seeding
+device = torch.device('cpu')
+random.seed(args.seed)
+np.random.seed(args.seed)
+torch.manual_seed(args.seed)
+torch.backends.cudnn.deterministic = args.torch_deterministic
+
+# MODIFIED: Env Wrapper for Observation and Rewards normalizations from the original repository
+from cleanrl.experiments.impl_matters.torch_utils import RunningStat, ZFilter, Identity, StateWithTime, RewardFilter
```

Comment:
We should probably move everything to this file since these optimizations are highly related to the algorithm.